consensus/polybft/consensus_runtime.go

```diff
@@ -208,6 +217,7 @@ func (c *consensusRuntime) OnBlockInserted(block *types.Block) {
 	} else {
 		c.lock.Lock()
 		c.lastBuiltBlock = block.Header
+		c.proposerCalculator.Update(block.Number(), c.config, c.state)
```

Comment:
It does not get called at the end of the epoch?

Reply:
It is called after each block. `OnBlockInserted` will be modified through https://github.com/0xPolygon/polygon-edge/pull/1047 which will be address possible race conditions and it is built on top of this PR.